devito/arch/compiler.py

```diff
@@ -528,6 +528,58 @@ def __lookup_cmds__(self):
         self.MPICXX = 'mpicxx'
 
 
+class OneapiCompiler(Compiler):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        language = kwargs.pop('language', configuration['language'])
+        platform = kwargs.pop('platform', configuration['platform'])
+
+        self.cflags.append("-xHost")
+        self.cflags.append("-qopt-zmm-usage=high")
+
+        if configuration['safe-math']:
+            self.cflags.append("-fp-model=strict")
+        else:
+            self.cflags.append('-fast')
+
+        if language == 'sycl':
+            self.cflags.append('-fsycl')
+            if platform is NVIDIAX:
+                self.cflags.append('-fsycl-targets=nvptx64-cuda')
+            else:
+                self.cflags.append('-fsycl-targets=spir64')
+
+        if language == 'openmp':
+            self.cflags.append('-fopenmp')
+            if platform is NVIDIAX:
+                self.cflags.append('-fopenmp-targets=nvptx64-cuda')
+            if platform is INTELGPUX:
+                self.cflags.append('-fopenmp-targets=spir64')
+                self.cflags.append('-fopenmp-target-simd')
+
+        if platform is INTELGPUX:
+            self.cflags.remove('-g')  # -g disables some optimizations in IGC
+            self.cflags.append('-gline-tables-only')
+            self.cflags.append('-fdebug-info-for-profiling')
+
+        # Make sure the MPI compiler uses `icx` underneath -- whatever the MPI distro is
+        if kwargs.get('mpi'):
+            ver = check_output([self.MPICC, "--version"]).decode("utf-8")
+            if not ver.startswith("Intel(R) oneAPI"):
```

Comment:
No here we should rather use `sniff_mpi_distro` (which potentially needs to be updated by adding "ver.startswith(Intel oneAPI "...")

And it's not just to avoid code duplication, but also because sniff_mpi_distro is memoized